### DISCUSSION TITLE
Uno SDK version change

### DISCUSSION THREAD

**Discussion by mhrstv** [unoplatform/uno#18811] [Q&A] [answered]
I tried changing my **Uno SDK** version. I ran into functionality problems.
What I was running was `Uno SDK 5.3.90`
#
Steps:
- Changed my **SDK** version in the `global.json` file from `5.3.90` to a higher one such as `5.4` **/** `5.5` (In my case it was `5.5.43`, and I also have previously tried with `5.4.10`)
- Changed `UnoExtensionsVersion` from `4.2.2` **->** `5.1.2`
- Changed `UnoToolkitVersion` from `6.1.8` **->** `6.3.6`
- Changed <ins>HotReload</ins> in `App.xaml.cs` to `UseStudio()`
#
- On `Desktop` (_Skia_), when I run the program it loads correctly. Everything seemed to be working fine, until I opened a _**Content Dialog**_.
- Content dialog buttons are _unusable_ / have _no_ functionality.
- On `Android` it works as intended and has no running issues.
#
Example videos of repro:
[SDK/Package version changes](https://streamable.com/r0cazn)
[Content Dialog Example](https://streamable.com/zdttxc)

**Comment by jeromelaban** [comment]
Thanks for the report. 

You probably created your solution a while back, and starting from Uno 5.4, you can remove all the versions that are specific in the Directory.Build.props, those are handled automatically by the .NET SDK. You only need to the global.json uno.sdk version.

Note when changing uno.sdk, you need to close and reload the solution. Recent versions of Uno show a banner that automatically restarts VS after an uno.sdk change.

For the `ContentDialog` issue, can you provide a repro sample of the issue? I'm not able to reproduce what you've shown at this time.

**Reply by mhrstv** [reply]
I did reload the solution and the IDE itself after changing the SDK version.
Also, I have sent samples of the `ContentDialog` repro and one during my SDK and package version changes.
You can see them in the bottom of the original discussion message.

**Reply by mhrstv** [reply]
As I said in the `#development-help` channel, I believe it would be better to migrate the old project files to a new one, running `Uno SDK 5.5`

**Reply by jeromelaban** [reply]
I don't see the repro sample in the links you posted. A repro sample [is as follows](https://platform.uno/docs/articles/uno-howto-create-a-repro.html).

Re-creating the project with 5.5 should not be required there aren't many changes that could cause this type of behavior.

**Reply by mhrstv** [reply]
[Repro video](https://streamable.com/qm7pwt)
[UnoApp1.zip](https://github.com/user-attachments/files/17780951/UnoApp1.zip)
- Here's a video + a new project with what I copied from my original project

**Reply by jeromelaban** [reply, marked as answer]
Thanks for the repro. `ContentDialog` is not supported when created inside a `ResourceDictionary`, the same is true when running WinAppSDK.

You must create a separate XAML file which contains your `ContentDialog`, uno provides a template for it in Visual Studio. You can find examples [here](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.controls.contentdialog?view=windows-app-sdk-1.6#examples).

**Reply by mhrstv** [reply]
Thanks!
How come it was running on Uno SDK 5.3 though?
Also, I'm not running WinAppSDK. Just a general question - it is intended only for MS Store apps, is that so ?

**Reply by mhrstv** [reply]
Ah, also, if I upgrade to Uno SDK 5.5, do I delete my package versions from the  `.props` file ?

**Reply by jeromelaban** [reply]
Yes, the versions are not needed, you only need the global.json uno.sdk version.
